### PR TITLE
Expose validation errors in the ApiException class

### PR DESCRIPTION
**Pull request by chrismhilton**
This change adds a method to return validation errors which can be used like the basic example below.

```
begin
  xero_invoice.save
rescue Xeroizer::ApiException => e
  if e.validation_errors.size > 0
    self.invoice.errors.add(:base, I18n.t(:msg_xero_invoice_failure, :errors => e.validation_errors.join(' ')))
  else
    self.invoice.errors.add(:base, I18n.t(:msg_xero_invoice_failure, :errors => e.message))
  end
end

```

### CODE REVIEW

File: lib/xeroizer/exceptions.rb [hunk 1]
```diff
@@ -28,6 +28,16 @@ def initialize(type, message, xml, parsed_xml, request_body)
     def message
       "#{@type}: #{@message} \n Generated by the following XML: \n #{@xml}"
     end
+
+    def validation_errors
+      errors = []
+      @parsed_xml.xpath("//ValidationError").each do |err|
+        errors = errors | [err.text.gsub(/^\s+/, '').gsub(/\s+$/, '')]
```
Comment (waynerobinson) [inline lib/xeroizer/exceptions.rb]: Does it need to be a union here, instead of a straight append?

Comment (chrismhilton) [inline lib/xeroizer/exceptions.rb]: I took that approach to prevent duplicate error messages that I encountered whilst testing

Comment (waynerobinson) [inline lib/xeroizer/exceptions.rb]: Why would there be duplicates? And if there were, don't they have some type of context that makes it important to show all of them?

**Comment by chrismhilton** [comment]
Hi Wayne.

Following your feedback I've changed the method so that it uses append rather than union when adding the errors to the array.

Cheers,
Chris

**Comment by waynerobinson** [comment]
Thanks for that. :)

**Pull request merged**